### PR TITLE
[ADD] New module ir_actions_report_xml_multi_company

### PR DESCRIPTION
**Pull request by JonathanNEMRY**

### CODE REVIEW

File: ir_actions_report_xml_multi_company/README.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+The usual multi-company record rule is installed automatically
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runboddt.odoo-community.org/runbot/133/10.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
+
+Known issues / Roadmap
```
Comment (lasley) [inline ir_actions_report_xml_multi_company/README.rst]: Remove empty section

File: ir_actions_report_xml_multi_company/README.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   :alt: License: AGPL-3
+
+===================================
+Ir Actions Report Xml Multi Company
+===================================
+
+Make Report Actions multi-company Aware
+
+Installation
+============
```
Comment (lasley) [inline ir_actions_report_xml_multi_company/README.rst]: Can remove installation & config sections if no instructions

File: ir_actions_report_xml_multi_company/README.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Usage
+=====
+
+The usual multi-company record rule is installed automatically
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runboddt.odoo-community.org/runbot/133/10.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (lasley) [inline ir_actions_report_xml_multi_company/README.rst]: Remove comment

File: ir_actions_report_xml_multi_company/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 ACSONE SA/NV, Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Ir Actions Report Xml Multi Company',
+    'summary': """
```
Comment (lasley) [inline ir_actions_report_xml_multi_company/__manifest__.py]: This multi line string is introducing errant whitespace. Please use single line string, concatenated if multi lines are necessary:

```
'summary': 'my long'
           'summary',
```

File: ir_actions_report_xml_multi_company/models/ir_actions_report_xml.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+    _inherit = 'ir.actions.report.xml'
+
+    @api.model
+    def _default_company(self):
+        return self.env['res.company']._company_default_get('res.partner')
+
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company', default=_default_company)
```
Comment (lasley) [inline ir_actions_report_xml_multi_company/models/ir_actions_report_xml.py]: `default=lambda s: s._default_company`

File: ir_actions_report_xml_multi_company/security/multi_company.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (lasley) [inline ir_actions_report_xml_multi_company/security/multi_company.xml]: `<odoo noupdate="1">` & can remove the data tag

File: ir_actions_report_xml_multi_company/views/ir_actions_report_xml.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+<data>
```
Comment (lasley) [inline ir_actions_report_xml_multi_company/views/ir_actions_report_xml.xml]: No need for data tag

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks @JonathanNEMRY - few suggestions

File: ir_actions_report_xml_multi_company/models/ir_actions_report_xml.py (hunk 5 above)
Comment (astirpe) [inline ir_actions_report_xml_multi_company/models/ir_actions_report_xml.py]: By installing this module on an already existing multicompany database, all the existing reports are automatically set with the default Company. This way all the existing reports become hidden for all the other companies. The point is that the most of the reports already present in the database are supposed to be used by all the companies; only few reports (eg. the invoices) are usually customized per company.

I would personally prefer to NOT define the default company at all.
When the module is installed, the administrator should update the per-company reports with the proper Company.

**Review by astirpe — CHANGES_REQUESTED** [review]
I find this module very useful.
Please notice my comment about the default value for the Company.

**Comment by andreparames** [comment]
Thanks @lasley and @astirpe for the review and suggestions. I've applied the changes suggested.

**Comment by astirpe** [comment]
The travis fails because of the flake8 error:

```
ir_actions_report_xml_multi_company/models/ir_actions_report_xml.py:5:1: F401 'odoo.api' imported but unused
```

**Comment by andreparames** [comment]
@astirpe thanks.

**Review by astirpe — APPROVED** [review]
Thank you!
Just a nice to have: is it ok to rename this module to "ir_actions_report_multi_company", in order to be consistent with  #80 ?

**Pull request closed without merging**